File: EduChemSuite.API/EduChemSuite.API/Helpers/QuestionTypeHelper.cs
namespace EduChemSuite.API.Helpers;

public static class QuestionTypeHelper
{
    public static bool IsDiagram(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("chemical structure") || desc.Contains("diagram") || desc.Contains("molecule");
    }

    public static bool IsAtomicStructure(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("atomic structure") || desc.Contains("bohr model");
    }

    public static bool IsChemicalEquation(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("chemical equation") || desc.Contains("equation balancing");
    }

    public static bool IsElectronConfig(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("electron configuration");
    }

    public static bool IsLewisStructure(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("lewis structure") || desc.Contains("lewis dot");
    }

    public static bool IsPeriodicTable(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("periodic table");
    }

    public static bool IsStoichiometry(string? questionTypeDescription)
    {
        if (string.IsNullOrEmpty(questionTypeDescription)) return false;
        var desc = questionTypeDescription.ToLowerInvariant();
        return desc.Contains("stoichiometry");
    }

    public static bool IsAutoGradedTool(string? questionTypeDescription)
    {
        return IsDiagram(questionTypeDescription)
            || IsAtomicStructure(questionTypeDescription)
            || IsChemicalEquation(questionTypeDescription)
            || IsElectronConfig(questionTypeDescription)
            || IsLewisStructure(questionTypeDescription)
            || IsPeriodicTable(questionTypeDescription)
            || IsStoichiometry(questionTypeDescription);
    }
}

File: EduChemSuite.UI/UI/src/app/_helpers/question-type.helper.ts
export function isDiagramQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('chemical structure') || desc.includes('diagram') || desc.includes('molecule');
}

export function isAtomicStructureQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('atomic structure') || desc.includes('bohr model');
}

export function isChemicalEquationQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('chemical equation') || desc.includes('equation balancing');
}

export function isElectronConfigQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('electron configuration');
}

export function isLewisStructureQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('lewis structure') || desc.includes('lewis dot');
}

export function isPeriodicTableQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('periodic table');
}

export function isStoichiometryQuestion(questionTypeDescription?: string | null): boolean {
  if (!questionTypeDescription) return false;
  const desc = questionTypeDescription.toLowerCase();
  return desc.includes('stoichiometry');
}

export function isInteractiveToolQuestion(questionTypeDescription?: string | null): boolean {
  return isDiagramQuestion(questionTypeDescription)
    || isAtomicStructureQuestion(questionTypeDescription)
    || isChemicalEquationQuestion(questionTypeDescription)
    || isElectronConfigQuestion(questionTypeDescription)
    || isLewisStructureQuestion(questionTypeDescription)
    || isPeriodicTableQuestion(questionTypeDescription)
    || isStoichiometryQuestion(questionTypeDescription);
}
